src/main/java/util/SocketChannelUtil.java

```diff
@@ -1,22 +1,26 @@
 package util;
 
-import api.ByteBufferSerializableObject;
-import enums.PacketType;
-import impl.ByteBufferPacket;
-import impl.gen.PacketTypeStorage;
-import impl.gen.model.HeaderHolder;
-
 import java.io.IOException;
 import java.net.InetSocketAddress;
 import java.nio.ByteBuffer;
 import java.nio.channels.SocketChannel;
 import java.util.Optional;
-import java.util.function.Supplier;
+
+import api.ByteBufferSerializableObject;
+import enums.PacketType;
+import impl.ByteBufferPacket;
+import impl.gen.model.HeaderHolder;
+import service.ByteBufferService;
+import service.PacketTypeService;
 
 //TODO - remove some serialize and deserialize methods, depends on what we need on current side (client/server).
-//TODO - add metadata to specification.
 
+/**
+ * Utility for read/write packets  from/to channel.
+ */
 public class SocketChannelUtil {
```

Comment:
In my opinion `...Util` class has to contain only static fields and static methods. But here I see dynamic fields and methods.

I think we should discuss class naming rules in details.

According to my experience:
Service - class that will have only one instance (singleton) and reusable.
Util or Utils - final class that contains only static final fields or/and static methods. It has private constructor to avoid instance creation.